File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        def count(n):
            cnt = 0
            
            while n:
                cnt+=(n&1)
                n>>=1
                
            return cnt
        
        
        result = []
        
        
        for i in range(n+1):
            result.append(count(i))
            
        return result
        

File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    int count(int n){
        int cnt = 0;
        
        for(int i=0;i<32;i++){
            cnt+=(n&1);
            n>>=1;
        }
        
        return cnt;
    }
    
    vector<int> countBits(int n) {
        vector<int> res;
        
        for(int i=0;i<=n;i++){
            res.emplace_back(count(i));
        }
        
        return res;
    }
};